android/src/main/java/com/polidea/blemulator/SimulatedAdapter.java

```diff
@@ -424,11 +428,15 @@ public void monitorCharacteristic(int characteristicIdentifier,
                                       OnEventCallback<Characteristic> onEventCallback,
                                       OnErrorCallback onErrorCallback) {
         Log.i(TAG, "monitorCharacteristic");
+        dartValueHandler.addCharacteristicsUpdatePublisher(onEventCallback);
+        dartMethodCaller.monitorCharacteristic(characteristicIdentifier, transactionId, onErrorCallback);
+
     }
 
     @Override
     public void cancelTransaction(String transactionId) {
         Log.i(TAG, "cancelTransaction");
+        dartMethodCaller.cancelTransaction(transactionId);
```

Comment:
Because cancelling monitoring is a different case than cancelling all transactions and it should be implemented with this task as it is a standalone feature. Foundations are the same but it wasn't expensive to use the code from another branch.